### PR TITLE
DB-9054: retrieve replication latency

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by jyuanca** [comment]
Please also review ee branch

File: splice_machine/src/main/java/com/splicemachine/replication/ReplicationSystemProcedure.java [hunk 1]
```diff
@@ -61,6 +62,19 @@
...
+    public static void GET_REPLICATION_PROGRESS(ResultSet[] resultSets) throws StandardException, SQLException {
+        ReplicationManager replicationManager = EngineDriver.driver().manager().getReplicationManager();
+        long ts = replicationManager.getReplicationProgress();
+        if (ts > 0) {
+            DateTime dateTime = new DateTime(ts);
+            resultSets[0] = ProcedureUtils.generateResult(
+                    "Replication progress", dateTime.toString());
+        }
+        else {
+            resultSets[0] = ProcedureUtils.generateResult("Error", "Information for available");
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/replication/ReplicationSystemProcedure.java]: typo in the result string, I guess it should be "Information not available".

File: splice_machine/src/main/java/com/splicemachine/replication/ReplicationSystemProcedure.java [hunk 2]
```diff
@@ -61,6 +62,19 @@
 
     private static Logger LOG = Logger.getLogger(ReplicationSystemProcedure.class);
 
+    public static void GET_REPLICATION_PROGRESS(ResultSet[] resultSets) throws StandardException, SQLException {
+        ReplicationManager replicationManager = EngineDriver.driver().manager().getReplicationManager();
+        long ts = replicationManager.getReplicationProgress();
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/replication/ReplicationSystemProcedure.java]: It seems that we assume only one replication slave cluster. Can this logic be extended in the future if we have  multiple slave clusters?

File: hbase_sql/src/main/java/com/splicemachine/hbase/SpliceReplicationSinkChore.java [hunk 3]
```diff
@@ -260,19 +264,29 @@ private void updateProgress() throws IOException, KeeperException, InterruptedEx
...
-                        long appliedPosition = replicationProgress.get(walName);
+                    byte[] colName = CellUtil.cloneQualifier(cell);
+                    if(Arrays.equals(colName, HBaseConfiguration.REPLICATION_SNAPSHOT_TSCOL_BYTES)){
+                        ts = Bytes.toLong(CellUtil.cloneValue(cell));
                         //if (LOG.isDebugEnabled()) {
+                        SpliceLogUtils.info(LOG, "Process snapshot take at %s", new DateTime(ts).toString());
+                        //}
+                    }
+                    else {
+                        String walName = Bytes.toString(CellUtil.cloneQualifier(cell));
```
Comment (yxia92) [inline hbase_sql/src/main/java/com/splicemachine/hbase/SpliceReplicationSinkChore.java]: The call to CellUtil.cloneQualifier(cell) here seems redundant, should we just use colName?

File: splice_machine/src/main/java/com/splicemachine/replication/ReplicationSystemProcedure.java (hunk 2 above)
Comment (jyuanca) [reply splice_machine/src/main/java/com/splicemachine/replication/ReplicationSystemProcedure.java]: This system procedure is run on a slave cluster

Comment (yxia92) [reply splice_machine/src/main/java/com/splicemachine/replication/ReplicationSystemProcedure.java]: I see. Thanks!

File: platform_it/src/test/java/com/splicemachine/test/SpliceTestPlatformConfig.java [hunk 4]
```diff
@@ -321,8 +321,8 @@ public static Configuration create(String hbaseRootDirUri,
         // Replication
         //
         config.setBoolean("replication.source.eof.autorecovery", true);
-        config.set("hbase.replication.source.service", "com.splicemachine.replication.SpliceReplication");
-        config.set("hbase.replication.sink.service", "com.splicemachine.replication.SpliceReplication");
+        //config.set("hbase.replication.source.service", "com.splicemachine.replication.SpliceReplication");
+        //config.set("hbase.replication.sink.service", "com.splicemachine.replication.SpliceReplication");
```
Comment (arnaud-splice) [inline platform_it/src/test/java/com/splicemachine/test/SpliceTestPlatformConfig.java]: Why is that necessary?

Comment (jyuanca) [reply platform_it/src/test/java/com/splicemachine/test/SpliceTestPlatformConfig.java]: Comment them out so that splice can be started w/o building ee branch, because SpliceReplication is only available on ee branch. Without commenting them out, the following command will fail
./start-splice-cluster -pcdh6.3.0

You have to always build with ee branch
start-splice-cluster -pcdh6.3.0,ee
